flytekit/core/inference.py

```diff
@@ -0,0 +1,196 @@
+        port: int = 8000,
+        cpu: int = 1,
+        gpu: int = 1,
+        mem: str = "20Gi",
+        shm_size: str = "16Gi",
+        env: Optional[dict[str, str]] = None,
+        hf_repo_ids: Optional[list[str]] = None,
+        lora_adapter_mem: Optional[str] = None,
+    ):
+        """
+        Initialize NIM class for managing a Kubernetes pod template.
+
+        :param image: The Docker image to be used for the model server container. Default is "nvcr.io/nim/meta/llama3-8b-instruct:1.0.0".
+        :param health_endpoint: The health endpoint for the model server container. Default is "v1/health/ready".
+        :param port: The port number for the model server container. Default is 8000.
+        :param cpu: The number of CPU cores requested for the model server container. Default is 1.
+        :param gpu: The number of GPU cores requested for the model server container. Default is 1.
+        :param mem: The amount of memory requested for the model server container. Default is "20Gi".
+        :param shm_size: The size of the shared memory volume. Default is "16Gi".
+        :param env: A dictionary of environment variables to be set in the model server container.
+        :param hf_repo_ids: A list of Hugging Face repository IDs for LoRA adapters to be downloaded.
+        :param lora_adapter_mem: The amount of memory requested for the init container that downloads LoRA adapters.
+        :param secrets: Instance of NIMSecrets for managing secrets.
+        """
+        if secrets.ngc_image_secret is None:
+            raise ValueError("NGC image pull secret must be provided.")
+        if secrets.ngc_secret_key is None:
+            raise ValueError("NGC secret key must be provided.")
+
+        self._shm_size = shm_size
+        self._hf_repo_ids = hf_repo_ids
+        self._lora_adapter_mem = lora_adapter_mem
+        self._secrets = secrets
+
+        super().__init__(
+            image=image,
+            health_endpoint=health_endpoint,
+            port=port,
+            cpu=cpu,
+            gpu=gpu,
+            mem=mem,
+            env=env,
+        )
+
+        self.setup_nim_pod_template()
+
+    def setup_nim_pod_template(self):
+        from kubernetes.client.models import (
+            V1Container,
+            V1EmptyDirVolumeSource,
+            V1EnvVar,
+            V1LocalObjectReference,
+            V1ResourceRequirements,
+            V1SecurityContext,
+            V1Volume,
+            V1VolumeMount,
+        )
+
+        self.pod_template.pod_spec.volumes = [
+            V1Volume(
+                name="dshm",
+                empty_dir=V1EmptyDirVolumeSource(medium="Memory", size_limit=self._shm_size),
+            )
+        ]
+        self.pod_template.pod_spec.image_pull_secrets = [V1LocalObjectReference(name=self._secrets.ngc_image_secret)]
+
+        model_server_container = self.pod_template.pod_spec.init_containers[0]
+
+        secret_prefix = get_plugin().secret_prefix()
```

Comment:
The configuration plugin system is designed to be global, but it is currently used for this plugin.

Can we hard code the prefix in `NIMSecrets` and use it from there?

Reply:
cc @eapolinario